src/sections/upload-dataset-files/uploaded-files-list/add-tags-modal/AddTagsModal.tsx

```diff
@@ -0,0 +1,135 @@
+import { Button, Col, Form, Modal, Badge } from '@iqss/dataverse-design-system'
+import styles from './AddTagsModal.module.scss'
+import { FormEvent, useState, KeyboardEvent, useId } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Plus, X } from 'react-bootstrap-icons'
+
+interface AddTagsModalProps {
+  show: boolean
+  availableTags: string[]
+  setTagOptions: (newTags: string[]) => void
+  update: (res: AddTagsModalResult) => void
+}
+
+export interface AddTagsModalResult {
+  saved: boolean
+  tags: string[]
+}
+
+export function AddTagsModal({ show, availableTags, setTagOptions, update }: AddTagsModalProps) {
+  const { t } = useTranslation('uploadDatasetFiles')
+  const [tagsToAdd, setTagsToAdd] = useState<string[]>([])
+  const [tag, setTag] = useState('')
```

Comment:
Its a little bit confusing the tag naming.
Maybe something like `newTag` or `newCustomTag`